lib/retrieve_title.rb

```diff
@@ -62,9 +63,17 @@ def self.fetch_title(url)
       else
         current = chunk
       end
+      if !encoding && content_type = _response['content-type']&.strip&.downcase
+        if content_type =~ /charset="?([a-z0-9_-]+)"?/
+          encoding = $1
```

Comment:
I tend to prefer `Regexp.last_match(1)` because it feels less like a weird Perl variable.